presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarHeaderAndMethods.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar.annotations;
+
+import com.facebook.presto.type.SqlType;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class ScalarHeaderAndMethods
+{
+    private final ScalarHeader header;
+    private final List<Method> methods;
+
+    public ScalarHeaderAndMethods(ScalarHeader header, List<Method> methods)
+    {
+        this.header = header;
+        this.methods = methods;
+    }
+
+    public static List<ScalarHeaderAndMethods> fromFunctionDefinitionClassAnnotations(Class annotated)
```

Comment:
I would rather move those methods to parser, made them private, and leave this class as plain POJO. It is easier to follow parser algorithm if all the methods are close to each other.